Source/missiles.cpp

```diff
@@ -1207,6 +1207,524 @@ void InitMissiles()
+		} else {
+			missile[mi]._miDelFlag = TRUE;
+		}
+	} else {
+		missile[mi]._miDelFlag = TRUE;
+	}
+}
+
+void missiles_stone_rune(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy, int id, int dam)
+{
+	if (LineClear(sx, sy, dx, dy)) {
+		if (id >= 0)
+			UseMana(id, 51); // TODO: apply enums
+		if (missiles_found_target(mi, &dx, &dy, 10)) {
+			missile[mi]._miVar1 = 30;
+			missile[mi]._miDelFlag = FALSE;
+			missile[mi]._mlid = AddLight(dx, dy, 8);
+		} else {
+			missile[mi]._miDelFlag = TRUE;
+		}
+	} else {
+		missile[mi]._miDelFlag = TRUE;
+	}
+}
+
+void missiles_reflection(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy, int id, int dam)
+{
+	int lvl;
+
+	if (id >= 0) {
+		if (missile[mi]._mispllvl)
+			lvl = missile[mi]._mispllvl;
+		else
+			lvl = 2;
+		plr[id].wReflection += lvl * plr[id]._pLevel;
+		UseMana(id, 43); // TODO: apply enums
+	}
+	missile[mi]._mirange = 0;
+	missile[mi]._miDelFlag = 0;
+}
+
+void missiles_berserk(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy, int id, int dam)
+{
+
+}
+
+void missiles_430624(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy, int id, int dam)
+{
+	GetMissileVel(mi, sx, sy, dx, dy, 8);
+	missile[mi]._mirange = 9;
+	missile[mi]._miVar1 = midir;
+	PutMissile(mi);
+}
+
+void missiles_jester(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy, int id, int dam)
+{
+	int spell;
+
+	spell = MIS_FIREBOLT;
+	switch (random(255, 10)) {
+	case 0:
+	case 1:
+		spell = MIS_FIREBOLT;
+		break;
+	case 2:
+		spell = MIS_FIREBALL;
+		break;
+	case 3:
+		spell = MIS_FIREWALLC;
+		break;
+	case 4:
+		spell = MIS_GUARDIAN;
+		break;
+	case 5:
+		spell = MIS_CHAIN;
+		break;
+	case 6:
+		spell = MIS_TOWN;
+		UseMana(id, SPL_TOWN);
+		break;
+	case 7:
+		spell = MIS_TELEPORT;
+		break;
+	case 8:
+		spell = MIS_APOCA;
+		break;
+	case 9:
+		spell = MIS_STONE;
+		break;
+	default:
+		break;
+	}
+	AddMissile(sx, sy, dx, dy, midir, spell, missile[mi]._micaster, missile[mi]._misource, 0, missile[mi]._mispllvl);
+	missile[mi]._miDelFlag = TRUE;
+	missile[mi]._mirange = 0;
+}
+
+void missiles_steal_pots(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy, int id, int dam)
+{
+
+}
+void missiles_mana_trap(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy, int id, int dam)
+{
+	int i, pn, k, j, tx, ty, pid;
+
+	missile[mi]._misource = id;
+	for (i = 0; i < 3; i++) {
+		k = CrawlNum[i];
+		pn = k + 2;
+		for (j = CrawlTable[k]; j > 0; j--) {
+			tx = sx + CrawlTable[pn - 1];
+			ty = sy + CrawlTable[pn];
+			if (0 < tx && tx < MAXDUNX && 0 < ty && ty < MAXDUNY) {
+				pid = dPlayer[tx][ty];
+				if (pid != 0) {
+					if (pid > 0)
+						pid = pid - 1;
+					else
+						pid = -(pid + 1);
+					plr[pid]._pMana = 0;
+					plr[pid]._pManaBase = plr[pid]._pMana + plr[pid]._pMaxManaBase - plr[pid]._pMaxMana;
+					CalcPlrInv(pid, FALSE);
+					drawmanaflag = TRUE;
+					PlaySfxLoc(246, tx, ty); //TODO: apply enum

```

Comment:
```suggestion
					PlaySfxLoc(TSFX_COW2, tx, ty);
```

Reply:
that's why I asked about it on discord :p